lib/usb_connection.js

```diff
@@ -175,12 +170,18 @@ USB.Connection.prototype._close = function(callback) {
   }
 
   self.closed = true;
-  self.intf.release(true, function() {
-    self.device.close();
-    if (typeof callback === 'function') {
-      callback();
-    }
-  }.bind(self));
+  self.epIn.stopPoll(function() {
+    self.intf.release(true, function attempt() {
+      try {
+        self.device.close();
+      } catch (e) {
+        setTimeout(attempt, 100);
+      }
+      if (typeof callback === 'function') {
+        callback();
+      }
+    }.bind(self));
```

Comment:
`bind(...)`isn't necessary if you already have a nasty `self` alias floating around ;)